### ISSUE TITLE
Marking A Notification Read

### ISSUE THREAD

**Issue by HonestAbe7** [Telegram-FOSS-Team/Telegram-FOSS#600]
Is it possible to mark a notification read by swiping left or right to dismiss it? I noticed that when I swipe them, they are still unread in Telegram.

**Comment by Younes-L** [comment]
TLDR : no, it is not possible to mark a notification read by swiping left or right to dismiss it.
There are alternatives (see below). Instead, if you need this to be a feature, keep in mind that [Telegram-FOSS](https://github.com/Telegram-FOSS-Team/Telegram-FOSS) (you are here) is not a feature fork : the goal of this fork is only to make an open source version of the original telegram app. This fork will not add arbitrary features, for that end you can ask telegram to add it here ([suggestions.telegram.org](https://suggestions.telegram.org)) or you can ask any of the feature forks to add it (such as [NekoX](https://github.com/nekoxdev/nekox) or [Forkgram](https://github.com/Forkgram/TelegramAndroid))

> Is it possible to mark a notification read

notifications in android cannot be marked as read, for they have no read/unread state.
You can however mark any chat in your telegram app as read. To do so, you can either : 
* open the chat and scroll to bottom, or
* select the chat and mark it as read, or
* receive a notification from the chat and tap the big blue `mark as read` button on the notification, before dismissing it (swiping left/right)
  * if you cannot see the `mark as read` button, expand the notification (by tapping the top part of the notification, next to the downwards arrow, or by swiping down with two fingers on the notification)
  * if you cannot expand the notification, then your OS removed essential android features, and you should either change your OS or look for another solution. If you need even more help, provide some information about your OS or device.

> by swiping left or right

swiping left or right on a notification will always dismiss the notification, on all android versions, on all devices. It will never do anything else.
So even though there's a `mark as read` button, the swipe action will still only dismiss the notification. The swipe action will not tap the `mark as read` button on your behalf : you must tap the button yourself.
Considering that the telegram app already has a feature to dismiss notifications on behalf of the user when the respective chats are marked as read, the reverse feature (mark chats as read on behalf of the user when the respective notifications are dimissed) seems feasible.

> I noticed that when I swipe them, they are still unread in Telegram.

yes, that is working as intended : notifications of your android system are one thing, chats in your telegram app/account are another.
Dismissing a notification will dismiss the notification. Marking a chat as read will mark the chat as read.
